src/cordova/apple/OutlineAppleLib/Sources/OutlineTunnelSources/OutlineTunnel.swift

```diff
@@ -41,26 +26,39 @@ public class OutlineTunnel: NSObject, Codable {
     case reconnecting = 2
   }
 
-  public convenience init(id: String, config: [String: Any]) {
+  public convenience init(id: String?, configString: String) {
     self.init()
     self.id = id
-    self.host = config["host"] as? String
-    self.password = config["password"] as? String
-    self.method = config["method"] as? String
-    if let port = config["port"] {
-      self.port = String(describing: port)  // Handle numeric values
-    }
-    if let prefix = config["prefix"] as? String {
-      self.prefix = Data(prefix.utf16.map{UInt8($0)})
-    }
+    self.configString = configString
+  }
+  
+  func encode() -> Data? {
+    return configString!.data(using: .utf8)
   }
 
-  public func encode() -> Data? {
-    return try? JSONEncoder().encode(self)
+  static func decode(_ encodedTunnelData: Data) -> OutlineTunnel? {
+    return OutlineTunnel(id: nil,
+                         configString: String(decoding: encodedTunnelData,
+                                              as: UTF8.self))
   }
 
-  public static func decode(_ jsonData: Data) -> OutlineTunnel? {
-    return try? JSONDecoder().decode(OutlineTunnel.self, from: jsonData)
+  // Private helper to retrieve the host from the config string.
+  private func configToDictionary() -> [String: Any]? {
```

Comment:
This means the config is not opaque to the platform-specific code, defeating the purpose of wiring the config.

What is the host used for besides the reachability test?

We may need to think about what we will show in the case of dynamic keys and what abstractions we need to support that. Perhaps we need to give different configs different ids. Or have the `Client` have a method to return the information to show.

Reply:
The action item here is to first understand what the host is being used for. We've talked about the Is Reachable: we should get rid of that. What else?